### PR TITLE
feat: remove deprecations from register page

### LINKED ISSUE #4237: Breaking Changes since v1.1.0: Register flow (#2799)
# Component
- [x] **Use below constructor:**
```
constructor(
    protected userService: UserService,
    protected globalMessageService: GlobalMessageService,
    protected fb: FormBuilder,
    protected router: RoutingService
  ) {}
```

- [x] **Remove variable:**
```
	isNewRegisterFlowEnabled: boolean
```

- [x] **Remove if statement:**
```
	if (this.isNewRegisterFlowEnabled) {...} else {...}
```
and use below code instead:
```
	this.loading$ = this.userService.getRegisterUserResultLoading();
    this.registerUserProcessInit();
```
# Unit tests
- [x] **Remove variable:**
```
	component.isNewRegisterFlowEnabled = true;
```

- [x] **Remove providers:**
```
	AuthRedirectService, AuthService, FeatureConfigService
```

### PR DESCRIPTION
**Pull request by kacperknapik**
Closes #4237

### CODE REVIEW

File: docs/migration/2_0.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ Default storage sync configuration was removed from `MultiCartStoreModule`. Inst
 
 Previously config validator logic was part of `ConfigModule`. If you are not using `StorefrontFoundationModule` or any of its descendants, it's required to import `ConfigValidatorModule.forRoot()` in order to make config validators run.
 
+### Deprecated since 1.1
+
+|  API  | Replacement |  Notes  |
+|-------|-------------|---------|
+| RegisterComponent(auth: AuthService, authRedirectService: AuthRedirectService, userService: UserService, globalMessageService: GlobalMessageService, fb: FormBuilder); | RegisterComponent(userService: UserService, globalMessageService: GlobalMessageService, fb: FormBuilder, router: RoutingService, anonymousConsentsService: AnonymousConsentsService, anonymousConsentsConfig: AnonymousConsentsConfig) | The constructor now uses new dependencies. `RegisterComponent(userService: UserService, globalMessageService: GlobalMessageService, fb: FormBuilder, router: RoutingService, anonymousConsentsService: AnonymousConsentsService, anonymousConsentsConfig: AnonymousConsentsConfig)` |
```
Comment (marlass) [inline docs/migration/2_0.md]: If you already mention that in automated changes below then there is no need to provide the information here. Let's keep these section for other changes (like method parameters change, or functio rename)

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  ANONYMOUS_CONSENTS_SERVICE,
+  ANONYMOUS_CONSENTS_CONFIG,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+// projects/storefrontlib/src/cms-components/user/register/register.component.ts
+export const REGISTER_COMPONENT_MIGRATIONS: ConstructorDeprecation[] = [
+  {
+    class: REGISTER_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    deprecatedParams: [
```
Comment (marlass) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts:22]: You have to specify in this section all current parameters. In your case, you need to add: UserService, GlobalMessageService and FormBuilder.

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  REGISTER_COMPONENT,
+  AUTH_SERVICE,
+  AUTH_REDIRECT_SERVICE,
+  ROUTING_SERVICE,
+  ANONYMOUS_CONSENTS_SERVICE,
+  ANONYMOUS_CONSENTS_CONFIG,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+// projects/storefrontlib/src/cms-components/user/register/register.component.ts
+export const REGISTER_COMPONENT_MIGRATIONS: ConstructorDeprecation[] = [
```
Comment (marlass) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts:18]: Also add second object to the config, where user provided all parameters (with RoutingService, FeatureConfigService, AnonymousConsentsService, AnonymousConsentsConfig), where you only specify removeParams with AuthService, AuthRedirectService and FeatureConfigService.

File: docs/migration/2_0.md [hunk 4]
```diff
@@ -79,3 +85,4 @@ Previously config validator logic was part of `ConfigModule`. If you are not usi
 - `CartItemComponent` now requires `PromotionService`. This service needs to be provided for `CartItemComponent`.
 - `CartCouponComponent` no longer uses `CartService`. This service usage was replaced with corresponding methods from `ActiveCartService`. Also `CustomerCouponService` and `FeatureConfigService` are now required parameters. These services needs to be provided for `CartCouponComponent`.
 - `CartItemListComponent` no longer uses `CartService`. This service usage was replaced with corresponding methods from `ActiveCartService.` Also `SelectiveCartService` and `FeatureConfigService` are now required parameters. These services needs to be provided for `CartItemListComponent`.
+- `RegisterComponent` no longer uses `AuthService` and `AuthRedirectService`. Also `RoutingService`, `AnonymousConsentsService` and `AnonymousConsentsConfig` are now required parameters.
```
Comment (marlass) [inline docs/migration/2_0.md]: Also mention that FeatureConfigService is no longer needed as well.

**Review by marlass — CHANGES_REQUESTED** [review]

File: docs/migration/2_0.md (hunk 1 above)
Comment (kacperknapik) [reply docs/migration/2_0.md]: Removed

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts (hunk 2 above)
Comment (kacperknapik) [reply projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts:22]: Added

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts (hunk 3 above)
Comment (kacperknapik) [reply projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts:18]: Done

File: docs/migration/2_0.md (hunk 4 above)
Comment (kacperknapik) [reply docs/migration/2_0.md]: Done

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      },
+      {
+        className: GLOBAL_MESSAGE_SERVICE,
+        importPath: SPARTACUS_CORE,
+      },
+      {
+        className: FORM_BUILDER,
+        importPath: ANGULAR_FORMS,
+      },
+      {
```
Comment (marlass) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts]: FeatureConfigService shouldn't be here.

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        className: ANONYMOUS_CONSENTS_SERVICE,
+        importPath: SPARTACUS_CORE,
+      },
+      {
+        className: ANONYMOUS_CONSENTS_CONFIG,
+        importPath: SPARTACUS_CORE,
+      },
+      {
```
Comment (marlass) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts:105]: Order of these must match order in constructor. Move FeatureConfigService

**Review by marlass — CHANGES_REQUESTED** [review]

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts (hunk 5 above)
Comment (kacperknapik) [reply projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts]: Fixed

File: projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts (hunk 6 above)
Comment (kacperknapik) [reply projects/schematics/src/migrations/2_0/constructor-deprecations/data/register.component.migration.ts:105]: Fixed

**Review by marlass — DISMISSED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**